### PR TITLE
requirements-doc: add PyYAML which removes dependency on -base (v2)

### PR DESCRIPTION
**Pull request by marc-hb**
This is "version 2" of #31239 rebased after #32593 which made many of the comments and discussions in #31239 obsolete.

~3 commits~ 2 commits, the main one:

This means a light requirements-doc.txt is enough for doc writers. See
previous discussions in PR #31199 and PR #31239

Also add comment that a west-less doc build is possible since the
spurious, indirect dependency on west was removed by commit
a56e1fa52d63a0 ("scripts: runners: bossac: handle edtlib ImportError")
which removed the `zephyr_ext_common` import from `bossac.py`.

Pro tip/note to self: changes to `bossac.py` do not trigger a fast, incremental doc build. Instead: `printf '\n' >> doc/guides/west/build-flash-debug.rst` does it (for bossac.py)

### CODE REVIEW

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -45,13 +45,7 @@ jobs:
     - name: install-pip
       run: |
         pip3 install setuptools wheel
-        pip3 install -r scripts/requirements-base.txt
         pip3 install -r scripts/requirements-doc.txt
-        pip3 install west==0.10.1
```
Comment (mbolivar-nordic) [inline .github/workflows/doc-build.yml:50]: NAK. We need west installed in the doc build workflow to get west API docs, and we need to control the exact version so the content of west-apis.rst matches.

File: .github/workflows/doc-build.yml [hunk 2]
```diff
@@ -45,13 +45,7 @@ jobs:
...
-
-    - name: west setup
-      run: |
-        west init -l . || true
```
Comment (mbolivar-nordic) [inline .github/workflows/doc-build.yml:54]: Are you sure about this one?

**Review by mbolivar-nordic — DISMISSED** [review]

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (marc-hb) [reply .github/workflows/doc-build.yml:50]: Then why does the doc build not fail without west, how's that possible?

Comment (mbolivar-nordic) [reply .github/workflows/doc-build.yml:50]: https://github.com/zephyrproject-rtos/zephyr/blob/master/doc/conf.py#L127

Comment (marc-hb) [reply .github/workflows/doc-build.yml:50]: Ah OK: the doc build supports both with or without west. Then shouldn't CI try both?

Comment (marc-hb) [reply .github/workflows/doc-build.yml:50]: Removed that bit

**Review by mbolivar-nordic — APPROVED** [review]

File: scripts/west_commands/runners/__init__.py [hunk 3]
```diff
@@ -12,10 +12,11 @@
...
         importlib.import_module(f'runners.{runner_name}')
-    except ImportError:
+    except ImportError as ie:
         # Runners are supposed to gracefully handle failures when they
         # import anything outside of stdlib, but they sometimes do
         # not. Catch ImportError to handle this.
+        _logger.warning(f'Caught ImportError: {ie}')
         _logger.warning(f'The module for runner "{runner_name}" '
                         'could not be imported. This most likely means '
                         'it is not handling its dependencies properly. '
```
Comment (gmarull) [inline scripts/west_commands/runners/__init__.py:22]: I'd suggest to merge changes into a single log, e.g.
```
        _logger.warning(f'The module for runner "{runner_name}" '
                        'could not be imported ({ie}). This most likely means '
                        'it is not handling its dependencies properly. '
```

File: .github/workflows/doc-build.yml [hunk 4]
```diff
@@ -45,8 +45,9 @@ jobs:
     - name: install-pip
       run: |
         pip3 install setuptools wheel
-        pip3 install -r scripts/requirements-base.txt
         pip3 install -r scripts/requirements-doc.txt
+        # Building without west is also possible, see exclude_patterns
+        # in conf.py
```
Comment (gmarull) [inline .github/workflows/doc-build.yml:50]: I don't think we want to build without west in CI, so this comment may be misleading here

**Review by gmarull — DISMISSED** [review]
6f65ddc7e99f9b1685a463fdd53ffb5dd37317d8 is not relevant anymore since the introduction of warnings_filter plugin.

File: .github/workflows/doc-build.yml (hunk 4 above)
Comment (marc-hb) [reply .github/workflows/doc-build.yml:50]: The "official" stance is that building the docs without west is not tested in CI but it is officially supported (and works last time I checked). So it is worth mentioning that one of the `requirements` is not actually a strict requirement.

File: scripts/west_commands/runners/__init__.py (hunk 3 above)
Comment (marc-hb) [reply scripts/west_commands/runners/__init__.py]: Done.

**Comment by marc-hb** [comment]
> 6f65ddc is not relevant anymore since the introduction of warnings_filter plugin.

I gave the github job yet another name because this (and other?) west warning(s) still escapes warnings_filter just like it escaped before.

**Comment by marc-hb** [comment]
@gmarull ping?

File: .github/workflows/doc-build.yml (hunk 4 above)
Comment (gmarull) [reply .github/workflows/doc-build.yml:50]: This information belongs to documentation build guidelines, but definitely not on CI scripts, where `west` is always desired as a requirement.

**Review by gmarull — DISMISSED** [review]
lgtm except CI comment.

File: .github/workflows/doc-build.yml (hunk 4 above)
Comment (marc-hb) [reply .github/workflows/doc-build.yml]: Done!

PS: when I want to build something not obvious I always look at CI scripts before looking at documentation; the latter is always lagging behind whereas the CI scripts are by nature always working. Plus this shortcut was not "officially supported", it'll never make it to the documentation.

**Review by gmarull — APPROVED** [review]

**Pull request merged**